packages/flutter_tools/lib/src/drive/drive_service.dart

```diff
@@ -218,22 +220,23 @@ class FlutterDriverService extends DriverService {
   Future<int> startTest(
     String testFile,
     List<String> arguments,
-    Map<String, String> environment, {
+    Map<String, String> environment,
+    PackageConfig packageConfig, {
     bool headless,
     String chromeBinary,
     String browserName,
     bool androidEmulator,
     int driverPort,
     List<String> browserDimension,
   }) async {
+    // Check if package:test is available. If not, fall back to invoking
+    // the test script directly.
     return _processUtils.stream(<String>[
       _dartSdkPath,
-      'pub',
-      'run',
-      'test',
-      ...arguments,
-      testFile,
-      '-rexpanded',
+      if (packageConfig['test'] != null)
```

Comment:
optional nit: Coming here with no context I'm asking myself in which cases do we need `pub run test` (e.g why not keep a single code path that invokes the test script directly), maybe worth writing in a comment.

Reply:
pub run test is strictly better, because in the even that you leave a socket or something open the VM will still correctly shut down instead of hang forever

Reply:
Can't we just have the plugins import package test?

Reply:
they don't need to import it, just depend on it

Reply:
We can (and sounds like we should), but I guess this is the way to not make it a breaking change?
If you're confident that this is only affecting flutter/plugins and after all affected packages have been migrated I guess it's safe to do the breaking change.

Reply:
I don't really want any more surprises this quarter, so independent of the changes to flutter/plugins I'd prefer to leave this in.